BootloaderCorePkg/Library/MpInitLib/Ia32/MpInitLib.c

```diff
@@ -75,14 +76,18 @@ CpuInit (
 
   if (PcdGet8 (PcdSmmRebaseMode) == SMM_REBASE_ENABLE_ON_S3_RESUME_ONLY) {
     if (GetBootMode() == BOOT_ON_S3_RESUME) {
-      SmmBaseInfo = (SMMBASE_INFO *) FindS3Info (SMMBASE_INFO_COMM_ID);
-      for (CpuIdx = 0; CpuIdx < SmmBaseInfo->SmmBaseHdr.Count; CpuIdx++) {
-        if (ApicId == SmmBaseInfo->SmmBase[CpuIdx].ApicId) {
-          SmmRebase (Index, ApicId, SmmBaseInfo->SmmBase[CpuIdx].SmmBase);
-          break;
+      CommHdr = (BL_PLD_COMM_HDR *) PcdGet32(PcdSmramTsegBase);
+      if (CommHdr->Signature == BL_PLD_COMM_SIG) {
+        SmmBaseInfo = (SMMBASE_INFO *) FindS3Info (SMMBASE_INFO_COMM_ID);
```

Comment:
we don't need use PcdSmramTsegBase, FindS3Info (SMMBASE_INFO_COMM_ID) should return NULL when SMMBASE_INFO  could not be found. so we need need check if NULL is returned here.